### PR TITLE
Feature/game logic

### PR DESCRIPTION
**Pull request by kbjude**
- The implementation of the game logic/ game Engine
- A few changes to the main file to relate with the game logic

### CODE REVIEW

File: lib/player.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Player
+  attr_reader :character
+  attr_reader :name
+  attr_accessor :score
```
Comment (kobiyoyo) [inline lib/player.rb:4]: Use attr_readers/attr_writers instead of attr_accessors

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, but there are a few more things required. 
### General
- [x] Lets fix Stickler errors
- [x] Note there should be a winner when the:
-  three horizontal elements  are equal
-  three vertical elements are equal 
- three equal elements form a diagonal, from left and right to right
- [x] This should all display the winner but am been asked to input.(below)



- [x] All classes are kept in `lib` directory. You have `UserInterface` class in main
- [x]   The code should respect the single responsibility rule for methods. Draw should have its own function,it should  not be  in ` move(player, num)`
Read comments below for more details
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: lib/player.rb (hunk 1 above)
Comment (meme-es) [reply lib/player.rb:4]: Could you tell us why to help us understand?

Comment (meme-es) [reply lib/player.rb:4]: By the way, the draw method is in game.board not in the move(player, num), so the code respect the single responsibility.

Comment (kobiyoyo) [reply lib/player.rb:4]: Using it  allow writing that property from outside the class. That variable can be set to anything

Comment (meme-es) [reply lib/player.rb:4]: That is the purpose, I understand the attribute accessor... I mean why 'use attr_reader/attr_writer instead of attr_accessor?'

Comment (kbjude) [reply lib/player.rb:4]: Kobiyoyo do you mean that we work on attr_accessor :score as attr_reader ? And if yes, could you advise according to the Manuel's question

**Comment by meme-es** [comment]
### Lint error

The lint error shows the next info, could a reviewer help us to know how to solve it? Thanks in advance!

Selection_005

**Review by Torres-ssf — APPROVED** [review]
### Project Approved 👍 
Hi there team ✋ 
No job implementing the requested changes. I tested your game and everything seems to work. 
- About the attr_acessor, I think for score it makes sense to be used since it needs to be both read and updated.
- About the linter errors: Rubocop added some new cops and you need to explicitly tell if they are going to be enabled or not. For now, we can list them as disabled.
Ex:
```
Lint/RaiseException:
  Enabled: false
```
Do this for all 5 and the linter will stop complaining. I checked you project locally and there aren't any other offences besides these ones, so **I'm approving this project**. But please, before merging, add these new rules to .rubocop.yml file and make a commit in order to make the linter stop complaining.

Good luck with the next milestone.
Happy coding 💻 

**Sergio Torres**

**Comment by meme-es** [comment]
Thanks for the help @Torres-ssf

**Pull request merged**